src/test/java/uk/gov/pay/products/service/ProductApiTokenManagerTest.java

```diff
@@ -0,0 +1,132 @@
+package uk.gov.pay.products.service;
+
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.LoggingEvent;
+import ch.qos.logback.core.Appender;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.products.client.publicauth.model.NewApiTokenFromPublicAuthResponse;
+import uk.gov.pay.products.config.ProductsConfiguration;
+import uk.gov.pay.products.exception.FailToRotateApiTokenException;
+import uk.gov.pay.products.exception.MissingApiTokenException;
+import uk.gov.pay.products.fixtures.ProductEntityFixture;
+import uk.gov.pay.products.model.Product;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import java.util.List;
+import java.util.Optional;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ProductApiTokenManagerTest {
+    private static final String PUBLICAUTH_URL = "https://publicauth.url";
+    private static final String EMAIL_ADDRESS_FOR_ROTATING_API_TOKENS = "pay-products@gov.uk";
+    private static final String NEW_API_TOKEN = "New API Token";
+
+    private ProductApiTokenManager productApiTokenManager;
+    private Product demoPayment = ProductEntityFixture.aProductEntity().build().toProduct();
+
+    @Mock
+    private Client client;
+
+    @Mock
+    private ProductsConfiguration productsConfiguration;
+
+    @Mock
+    private ProductFactory productFactory;
+
+    private Appender<ILoggingEvent> mockAppender;
+
+    @Captor
+    private ArgumentCaptor<LoggingEvent> loggingEventArgumentCaptor;
+
+    @Before
+    public void setUp() {
+        Logger root = (Logger) LoggerFactory.getLogger(ProductApiTokenManager.class);
+        mockAppender = mock(Appender.class);
```

Comment:
Is there something about the fact this has to be added to `root` that stops us using the `@Mock` annotation for this?